impeller/renderer/backend/metal/allocator_mtl.mm

```diff
@@ -173,4 +173,31 @@ static MTLStorageMode ToMTLStorageMode(StorageMode mode,
   return std::make_shared<TextureMTL>(desc, texture);
 }
 
+ISize AllocatorMTL::GetMaxTextureSizeSupported() {
+  // Since Apple didn't expose API for us to get the max texture size, we have
+  // to use hardcoded data from
+  // https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf
+  // According to the feature set table, there are two supported max sizes :
+  // 16384 and 8192 for devices flutter support. The former is used on macs and
+  // latest ios devices. The latter is used on old ios devices.
+  if (@available(macOS 10.15, iOS 13, tvOS 13, *)) {
```

Comment:
Really not sure how expensive the determination of these limits is, but perhaps figure these out in the ctor of the allocator? See the [UMA and memoryless targets support check here](https://github.com/flutter/engine/blob/2c3efa0beb248c6afd4a0a5eb66ea3fb67a1f9ed/impeller/renderer/backend/metal/allocator_mtl.mm#L59).

Reply:
Done.